### PR TITLE
Improvement/definition

### PR DESCRIPTION
**Pull request by Zorin95670**
:information_source:  Don't forget to modify the changelog.md before merging this branch.

### CODE REVIEW

File: src/models/ComponentAttribute.js [hunk 1]
```diff
@@ -8,14 +8,21 @@ class ComponentAttribute {
    * @param {String} [props.name=null] - The name of the attribute.
```
Comment (Zorin95670) [inline src/models/ComponentAttribute.js:8]: Missing changelog

File: src/models/ComponentAttribute.js [hunk 2]
```diff
@@ -8,14 +8,21 @@ class ComponentAttribute {
...
    * the attribute.
-   * @param {String} [props.type=null] - The type of the attribute.
+   * @param {String} [props.type=null] - The reel type of the attribute.
+   * @param {ComponentAttributeDefinition} [props.definition=null] - The definition of the
+   * attribute.
    */
   constructor(props = {
     name: null,
     value: null,
-    type: null,
```
Comment (k-y0u) [inline src/models/ComponentAttribute.js:16]: why `type` disappear ?

File: tests/unit/models/ComponentAttribute.spec.js [hunk 3]
```diff
@@ -16,18 +17,21 @@ describe('Test class: ComponentAttribute', () => {
...
         name: 'name',
         value: 'value',
         type: 'type',
+        definition: {},
       });
 
       expect(componentAttribute.name).toEqual('name');
       expect(componentAttribute.value).toEqual('value');
       expect(componentAttribute.type).toEqual('type');
+      expect(componentAttribute.definition).not.toBeNull();
```
Comment (k-y0u) [inline tests/unit/models/ComponentAttribute.spec.js:34]: `toBeDefined` ?

File: src/models/ComponentLink.js [hunk 4]
```diff
@@ -8,38 +8,36 @@ class ComponentLink extends FileInformation {
   /**
    * Default constructor.
    *
-   * @param {Object} [props.from={}] - Where the link is created.
-   * @param {String} [props.from.id] - ID of the component.
-   * @param {Object} [props.to={}] - Target of the link.
-   * @param {String} [props.to.id] - ID of the component.
-   * @param {String} [props.type] - Representation of the link.
+   * @param {String} [props.sourceRef] - Id of component can be the source in a link.
+   * @param {String} [props.targetRef] - Id of component can be the target of the link.
```
Comment (Hundraw) [inline src/models/ComponentLink.js:12]: why sourceRef and targetRef here and not everywhere we found it?

File: tests/unit/models/ComponentLink.spec.js [hunk 5]
```diff
@@ -5,37 +5,29 @@ describe('Test class: ComponentLink', () => {
...
+        target: 'target',
+        definition: {},
       });
 
-      expect(link.from.id).toEqual('id1');
-      expect(link.to.id).toEqual('id2');
-      expect(link.type).toEqual('type');
+      expect(link.source).toEqual('source');
+      expect(link.target).toEqual('target');
+      expect(link.definition).not.toBeNull();
```
Comment (k-y0u) [inline tests/unit/models/ComponentLink.spec.js:30]: same here

File: changelog.md [hunk 6]
```diff
@@ -5,6 +5,20 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)
 
+## [Unreleased]
+
+### Added
+
+- Add definition in `ComponentAttribute`
```
Comment (k-y0u) [inline changelog.md:12]: `definition` isn't  a property too ?

**Review by k-y0u — CHANGES_REQUESTED** [review]

**Review by Hundraw — CHANGES_REQUESTED** [review]

File: src/models/ComponentAttribute.js (hunk 2 above)
Comment (Zorin95670) [reply src/models/ComponentAttribute.js:16]: mistake, done :)

File: tests/unit/models/ComponentAttribute.spec.js (hunk 3 above)
Comment (Zorin95670) [reply tests/unit/models/ComponentAttribute.spec.js:34]: done

File: tests/unit/models/ComponentLink.spec.js (hunk 5 above)
Comment (Zorin95670) [reply tests/unit/models/ComponentLink.spec.js:30]: done

File: changelog.md (hunk 6 above)
Comment (Zorin95670) [reply changelog.md:12]: done

File: src/models/ComponentLink.js (hunk 4 above)
Comment (Zorin95670) [reply src/models/ComponentLink.js]: done

**Review by k-y0u — APPROVED** [review]

**Review by Hundraw — APPROVED** [review]

**Pull request merged**